packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1835,6 +1849,18 @@ export class ContainerRuntime
 	public process(messageArg: ISequencedDocumentMessage, local: boolean) {
 		this.verifyNotClosed();
 
+		const ungroupingResult = this.opGroupingManager.processOp(messageArg);
+		if (ungroupingResult !== undefined) {
+			for (const subMessage of ungroupingResult) {
```

Comment:
It feels like it asks for processCore() or something of that nature, where process() will call number of times processCore() (and thus will not allow recursion, even if it will never be materialized)